### PR TITLE
Add "entity" class DockerHubContainer

### PR DESCRIPTION
**Pull request by Ichimonji10**
Class `DockerHubContainer` represents a container whose image comes from the
docker hub. This class is manually tested. It is possible to create, read and
delete containers. It is also possible to start, stop, get the status of, and
get logs from a container.

Be warned that the `read` method will not work until #1941 is merged.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Many thanks to @omaciel for helping me develop this.

File: robottelo/entities.py [hunk 1]
```diff
@@ -418,6 +419,143 @@ def create_payload(self):
...
+            verify=False,
+            data={u'power_action': power_action},
+        )
+        response.raise_for_status()
+        return response.json()
+
+    def logs(self, stdout=None, stderr=None, tail=None):
+        """Get logs from this container.
+
+        :param bool stdout: ???
```
Comment (omaciel) [inline robottelo/entities.py]: Not a blocker but let's have someone from the Docker Feature team look at this and give us some info?

File: tests/robottelo/test_entities.py [hunk 2]
```diff
@@ -47,6 +47,8 @@ def test_path_without_which(self, entity, path):
         )
 
     @data(
+        (entities.AbstractDockerContainer, 'containers', 'logs'),
```
Comment (omaciel) [inline tests/robottelo/test_entities.py]: Thank you for adding some tests!!!

**Comment by omaciel** [comment]
ACK and a question: will a ``class DockerLocalContainer`` be in the works soon? :)

File: robottelo/entities.py (hunk 1 above)
Comment (Ichimonji10) [inline robottelo/entities.py]: I agree.

**Comment by Ichimonji10** [comment]
@omaciel, I've stuffed some code for `class DockerLocalContainer` into a branch. You can see it [here](https://github.com/SatelliteQE/robottelo/compare/master...Ichimonji10:container-api-2). I may need to have a follow-up session with you to get that done. The session we had today was enormously productive (and exhausting!). :)

**Comment by elyezer** [comment]
ACK

**Pull request merged**